May 29 exec doc (#5)
makerdao/executive-votes: 2025/executive-vote-2025-05-29-MKR-to-SKY-upgrade-phase-two.md

This executive proposal includes an office-hours modifier that means that it **can only be executed between 14:00 and 21:00 UTC, Monday - Friday**. 

If this executive proposal does not pass within 30 days, then it will expire and can no longer have any effect on the Sky Protocol.

---

## Proposal Details

### MKR to SKY Upgrade Phase Two

- **Authorization**: [Atlas](https://sky-atlas.powerhouse.io/A.4.1.2.1.2_MKR_To_SKY_Upgrade_Phase_Two/1f1f2ff0-8d73-80ee-81e9-f508c45f90ed|b341f4c0b83472dc), [Governance Poll 1245](https://vote.makerdao.com/polling/QmcZNZg3)  
- **Proposal**: [Forum Post](https://forum.sky.money/t/atlas-edit-weekly-cycle-proposal-week-of-may-5-2025/26319)

> If this executive proposal passes, the following actions will complete Phase Two of the MKR to SKY Upgrade.

Reviewer: duplicate suggestion. resolving

#### Revoke Migrator Authority Over Vat Contract

- Revoke LOCKSTAKE_MIGRATOR's authority over the vat.

### Change the Source of Sky Token Rewards (SKY to USDS) to the Pause Proxy.

- **Authorization**: [Ecosystem Approval](https://forum.sky.money/t/proposed-housekeeping-item-upcoming-executive-spell-2025-05-29/26448/5)
- **Proposal**: [Forum Post](https://forum.sky.money/t/proposed-housekeeping-item-upcoming-executive-spell-2025-05-29/26448)

If this executive proposal passes, then a new vesting stream will be initialized for Sky Token Rewards. This stream will be adjusted to match SSR by changing the rewards distribution rate and its source of SKY will be switched from minting new tokens to transferring tokens from the PauseProxy. These changes are achieved by executing the following actions:

- `yank` MCD_VEST_SKY stream ID 2.
- Claim the remaining tokens from the old DssVest by calling VestedRewardsDistribution.distribute() on REWARDS_DIST_USDS_SKY
- Set cap on MCD_VEST_SKY_TREASURY to `151,250,000 * WAD / 182 days`.

> - Increase sky.approve(MCD_VEST_SKY_TREASURY, ...) by 137,500,000 SKY to account for new vesting stream.

Reviewer: ```suggestion
- Increase `sky.approve(MCD_VEST_SKY_TREASURY, ...)` by 137.5 million SKY to account for new vesting stream.
```

> - Increase sky.approve(MCD_VEST_SKY_TREASURY, ...) by 137,500,000 SKY to account for new vesting stream.

Reviewer: fixed separately in https://github.com/makerdao/executive-votes/pull/5/commits/2471c22715f7427c5e3a722238ee34a3a2609a80

> - Increase sky.approve(MCD_VEST_SKY_TREASURY, ...) by 137,500,000 SKY to account for new vesting stream.

Reviewer: fixed separately in https://github.com/makerdao/executive-votes/pull/5/commits/2471c22715f7427c5e3a722238ee34a3a2609a80

- Remove old REWARDS_DIST_USDS_SKY from the keeper job by calling `CRON_REWARDS_DIST_JOB.rem()` with the current REWARDS_DIST_USDS_SKY. 
- Update chainlog value for REWARDS_DIST_USDS_SKY to the new VestedRewardsDistribution contract at [0xC8d67Fcf101d3f89D0e1F3a2857485A84072a63F](https://etherscan.io/address/0xC8d67Fcf101d3f89D0e1F3a2857485A84072a63F).

> - Add new REWARDS_DIST_USDS_SKY to the keeper job by calling `CRON_REWARDS_DIST_JOB.set()` with the new REWARDS_DIST_USDS_SKY.

Reviewer 2: ```suggestion
- Add new REWARDS_DIST_USDS_SKY to the keeper job by calling `CRON_REWARDS_DIST_JOB.set()` with the new REWARDS_DIST_USDS_SKY  and frequency 601200.
```

Reviewer: fixed separately in https://github.com/makerdao/executive-votes/pull/5/commits/2471c22715f7427c5e3a722238ee34a3a2609a80

> - Add new REWARDS_DIST_USDS_SKY to the keeper job by calling `CRON_REWARDS_DIST_JOB.set()` with the new REWARDS_DIST_USDS_SKY.

Reviewer: ```suggestion
- Add new REWARDS_DIST_USDS_SKY to the keeper job by calling `CRON_REWARDS_DIST_JOB.set()` with the new REWARDS_DIST_USDS_SKY and frequency of 601,200 seconds.
```

### Transfer Ownership of SPK Token to SPK Company Multisig

- **Authorization**: [Ecosystem Approval](https://forum.sky.money/t/proposed-housekeeping-item-upcoming-executive-spell-2025-05-29/26448/5)
- **Proposal**: [Forum Post](https://forum.sky.money/t/proposed-housekeeping-item-upcoming-executive-spell-2025-05-29/26448/3)

If this executive proposal passes, then ownership of SPK Token will be transferred to the SPK Company Multisig. This is achieved through the following actions:

- Rely [0x6FE588FDCC6A34207485cc6e47673F59cCEDF92B](https://etherscan.io/address/0x6FE588FDCC6A34207485cc6e47673F59cCEDF92B) on [0xc20059e0317DE91738d13af027DfC4a50781b066](https://etherscan.io/address/0xc20059e0317DE91738d13af027DfC4a50781b066).
- Deny MCD_PAUSE_PROXY on [0xc20059e0317DE91738d13af027DfC4a50781b066](https://etherscan.io/address/0xc20059e0317DE91738d13af027DfC4a50781b066).

### Increase ALLOCATOR-SPARK-A Maximum Debt Ceiling

- **Authorization**: [Governance Poll 1251](https://vote.makerdao.com/polling/QmVMBXm8)
- **Proposal**: [Forum Post](https://forum.sky.money/t/may-29-2025-proposed-changes-to-spark-for-upcoming-spell/26372)

> If this executive proposal passes, then the ALLOCATOR-SPARK-A [Maximum Debt Ceiling (`line`)](https://sky-atlas.powerhouse.io/A.3.8.1.1.2.4.1_Maximum_Debt_Ceiling_(line)/6f1a913d-9436-4b70-816b-e317672737d6%7C57eaf45219bea3b430c268bb) will be increased by 5 billion USDS from 5 billion USDS to **10 billion USDS**. The `gap` and `ttl` parameters for ALLOCATOR-SPARK-A will remain unchanged. 

Reviewer 2: Since these parameters have to be included in the spell even if they haven't changed, I'd include the figures in the Exec Doc as well

Reviewer: added in https://github.com/makerdao/executive-votes/pull/5/commits/04d5105dede54c9b4f9cc6c5e53df90a4746f896

- L2_SUSDS: [0xA06b10Db9F390990364A3984C04FaDf1c13691b5](https://uniscan.xyz/address/0xA06b10Db9F390990364A3984C04FaDf1c13691b5)

- L1_SSR_FORWARDER: [0x45d91340B3B7B96985A72b5c678F7D9e8D664b62](https://etherscan.io/address/0x45d91340B3B7B96985A72b5c678F7D9e8D664b62)
- L2_SSR_RECEIVER: [0x4A71f81C6109230932978bAB7CA746f0be0C4580](https://uniscan.xyz/address/0x4A71f81C6109230932978bAB7CA746f0be0C4580)
- L2_SSR_AUTH_ORACLE: [0x1566BFA55D95686a823751298533D42651183988](https://uniscan.xyz/address/0x1566BFA55D95686a823751298533D42651183988)

- L2_GOV_RECEIVER: [0x7B8ee8b0fD62662F7FB1ac9e5E6cEAad5195A3bF](https://uniscan.xyz/address/0x7B8ee8b0fD62662F7FB1ac9e5E6cEAad5195A3bF)
- L2_GOV_EXECUTOR: [0xb037C43b433964A2017cd689f535BEb6B0531473](https://uniscan.xyz/address/0xb037C43b433964A2017cd689f535BEb6B0531473)

- L2_PSM: [0x7b42Ed932f26509465F7cE3FAF76FfCe1275312f](https://uniscan.xyz/address/0x7b42Ed932f26509465F7cE3FAF76FfCe1275312f)

- L2_ALM_PROXY: [0x345E368fcCd62266B3f5F37C9a131FD1c39f5869](https://uniscan.xyz/address/0x345E368fcCd62266B3f5F37C9a131FD1c39f5869)

> - L2_ALM_CONTROLLER: [0x9B1BEB11CFE05117029a30eb799B6586125321FF](https://uniscan.xyz/address/0x9B1BEB11CFE05117029a30eb799B6586125321FF?tab=contract)

Reviewer 3: ```suggestion
- L2_ALM_CONTROLLER: [0x9B1BEB11CFE05117029a30eb799B6586125321FF](https://uniscan.xyz/address/0x9B1BEB11CFE05117029a30eb799B6586125321FF)
```

(If you want to go to the contract tab, seems like you need to add "?tab=contract#code")

#### Increase USDS Mint and USDC Swap Rate Limits

- **Authorization**: [Governance Poll 1250](https://vote.makerdao.com/polling/QmUYRV1H)
- **Proposal**: [Forum Post](https://forum.sky.money/t/may-29-2025-proposed-changes-to-spark-for-upcoming-spell/26372)

Increase the Mint and Swap [Rate Limits](https://docs.spark.fi/dev/spark-liquidity-layer/spark-alm-controller#rate-limits) for USDS and USDC as follows.

##### USDS Mint Rate Limits

- Increase max by 300 million USDS from 200 million USDS to **500 million USDS**.
- Increase slope by 300 million USDS per day from 200 million USDS per day to **500 million USDS per day**.

##### USDC Swap Rate Limits

- Increase max by 300 million USDC from 200 million USDC to **500 million USDC**.
- Increase slope by 100 million USDC from 200 million USDC to **300 million USDC per day**.

#### Onboard August PT-USDS to Morpho Spark DAI Vault

- **Authorization**: [Governance Poll 1505](https://vote.sky.money/polling/QmRtLuL2)

> - **Proposal**: [Forum Post](https://forum.sky.money/t/may-29-2025-proposed-changes-to-spark-for-upcoming-spell/26372)

Reviewer 2: ```suggestion
- **Proposal**: [Forum Post](https://forum.sky.money/t/may-29-2025-proposed-changes-to-spark-for-upcoming-spell-2/26440)
```

- Increase sky.approve(MCD_VEST_SKY_TREASURY, ...) by 137,500,000 SKY to account for new vesting stream.
- Remove old REWARDS_USDS_SKY_DIST from the keeper job by calling `CRON_REWARDS_DIST_JOB.rem()` with the current REWARDS_USDS_SKY_DIST. 
- Update chainlog value for REWARDS_USDS_SKY_DIST to the new VestedRewardsDistribution contract at [0xC8d67Fcf101d3f89D0e1F3a2857485A84072a63F](https://etherscan.io/address/0xC8d67Fcf101d3f89D0e1F3a2857485A84072a63F).
- Add new REWARDS_USDS_SKY_DIST to the keeper job by calling `CRON_REWARDS_DIST_JOB.set()` with the new REWARDS_USDS_SKY_DIST.
- Deploy new MCD_VEST_SKY_TREASURY stream with the following parameters:
    - usr: REWARDS_USDS_SKY_DIST
    - tot: 137,500,000 SKY
    - bgn:  block.timestamp
    - tau: 182 days
    - cliff: none
    - manager: none.
- Restrict the new stream, res: 1.
- File the id of the newly created stream to the new REWARDS_USDS_SKY_DIST contract.

### Init Unichain Native Bridge

- **Authorization**: [Governance Poll 1253](https://vote.makerdao.com/polling/QmXjeJtw)
- **Proposal**: [Forum Post](https://forum.sky.money/t/may-29-2025-proposed-changes-to-spark-for-upcoming-spell/26372)

> If this executive proposal passes, then the Unichain Token Bridge for USDS and sUSDS will be initialized by calling [TokenBridgeInit.initBridges]([TokenBridgeInit.initBridges](https://github.com/makerdao/op-token-bridge/blob/82918f4853d50c6520dac53fdb70a42fd4ce671b/script/Init.s.sol)) using the following parameters.

Reviewer: Link directly to the function

```suggestion
If this executive proposal passes, then the Unichain Token Bridge for USDS and sUSDS will be initialized by calling [TokenBridgeInit.initBridges](https://github.com/makerdao/op-token-bridge/blob/0f935505c0dc74ce3db2a9998320a56119321814/deploy/TokenBridgeInit.sol#L63) using the following parameters.
```

- Set parameter cfg.l1Tokens: (USDS, SUSDS on mainnet)
- Set parameter cfg.l2Tokens: (USDS: [0x7E10036Acc4B56d4dFCa3b77810356CE52313F9C](https://uniscan.xyz/address/0x7E10036Acc4B56d4dFCa3b77810356CE52313F9C), sUSDS: [0xA06b10Db9F390990364A3984C04FaDf1c13691b5](https://uniscan.xyz/address/0xA06b10Db9F390990364A3984C04FaDf1c13691b5) on Unichain)
- Set parameter cfg.maxWithdraws: (type(uint256).max for each token)
- Set parameter cfg.minGasLimit: (500,000)
- Set parameter cfg.govRelayCLKey: (chainlog key for govRelay -> UNICHAIN_GOV_RELAY)
- Set parameter cfg.escrowCLKey: (chainlog key for Escrow -> UNICHAIN_ESCROW)
- Set parameter cfg.l1BridgeCLKey: (chainlog key for L1TokenBridge -> UNICHAIN_TOKEN_BRIDGE)
- Set parameter cfg.l1BridgeImpCLKey: (chainlog key for L1TokenBridgeImp -> UNICHAIN_TOKEN_BRIDGE_IMP)

### Init Optimism Native Bridge

- **Authorization**: [Governance Poll 1254](https://vote.makerdao.com/polling/QmNe8Erm)
- **Proposal**: [Forum Post](https://forum.sky.money/t/may-29-2025-proposed-changes-to-spark-for-upcoming-spell/26372)

> If this executive proposal passes, then the Unichain Token Bridge for USDS and sUSDS will be initialized by calling [TokenBridgeInit.initBridges](https://github.com/makerdao/op-token-bridge/blob/0f935505c0dc74ce3db2a9998320a56119321814/deploy/TokenBridgeInit.sol#L63) using the following parameters.

Reviewer 4: ```suggestion
If this executive proposal passes, then the Optimism Token Bridge for USDS and sUSDS will be initialized by calling [TokenBridgeInit.initBridges](https://github.com/makerdao/op-token-bridge/blob/0f935505c0dc74ce3db2a9998320a56119321814/deploy/TokenBridgeInit.sol#L63) using the following parameters.
```

> If this executive proposal passes, then the Optimism Token Bridge for USDS and sUSDS will be initialized by calling [TokenBridgeInit.initBridges](https://github.com/makerdao/op-token-bridge/blob/82918f4853d50c6520dac53fdb70a42fd4ce671b/script/Init.s.sol) using the following parameters.

Reviewer: Link directly to the function

```suggestion
If this executive proposal passes, then the Unichain Token Bridge for USDS and sUSDS will be initialized by calling [TokenBridgeInit.initBridges](https://github.com/makerdao/op-token-bridge/blob/0f935505c0dc74ce3db2a9998320a56119321814/deploy/TokenBridgeInit.sol#L63) using the following parameters.
```

#### Set l1BridgeInstance 

- Set parameter l1BridgeInstance.govRelay: (L1GovernanceRelay: [0x09B354CDA89203BB7B3131CC728dFa06ab09Ae2F](https://etherscan.io/address/0x09B354CDA89203BB7B3131CC728dFa06ab09Ae2F))
- Set parameter l1BridgeInstance.escrow: (Escrow: [0x467194771dAe2967Aef3ECbEDD3Bf9a310C76C65](https://etherscan.io/address/0x467194771dAe2967Aef3ECbEDD3Bf9a310C76C65))
- Set parameter l1BridgeInstance.bridge: (ERC1967Proxy: [0x3d25B7d486caE1810374d37A48BCf0963c9B8057](https://etherscan.io/address/0x3d25B7d486caE1810374d37A48BCf0963c9B8057))
- Set parameter l1BridgeInstance.bridgeImp: (L1TokenBridge: [0xA50adBad34c1e9786979bD44220F8fd46e43A6B0](https://etherscan.io/address/0xA50adBad34c1e9786979bD44220F8fd46e43A6B0))

#### Set l2BridgeInstance (Optimism Network)

> - Set parameter l2BridgeInstance.govRelay: (L2GovernanceRelay: [0x10E6593CDda8c58a1d0f14C5164B376352a55f2F](https://optimistic.etherscan.io/address/0x10E6593CDda8c58a1d0f14C5164B376352a55f2F)

Reviewer 4: ```suggestion
- Set parameter l2BridgeInstance.govRelay: (L2GovernanceRelay: [0x10E6593CDda8c58a1d0f14C5164B376352a55f2F](https://optimistic.etherscan.io/address/0x10E6593CDda8c58a1d0f14C5164B376352a55f2F))
```

- Set parameter l2BridgeInstance.bridge: (ERC1967Proxy: [0x8F41DBF6b8498561Ce1d73AF16CD9C0d8eE20ba6](https://optimistic.etherscan.io/address/0x8F41DBF6b8498561Ce1d73AF16CD9C0d8eE20ba6))

> - Set parameter l2BridgeInstance.bridgeImp: (L2TokenBridge: [0xc2702C859016db756149716cc4d2B7D7A436CF04](https://optimistic.etherscan.io/address/0xc2702C859016db756149716cc4d2B7D7A436CF04)

Reviewer 4: ```suggestion
- Set parameter l2BridgeInstance.bridgeImp: (L2TokenBridge: [0xc2702C859016db756149716cc4d2B7D7A436CF04](https://optimistic.etherscan.io/address/0xc2702C859016db756149716cc4d2B7D7A436CF04))
```

- L2_SSR_RECEIVER: [0x4A71f81C6109230932978bAB7CA746f0be0C4580](https://uniscan.xyz/address/0x4A71f81C6109230932978bAB7CA746f0be0C4580)
- L2_SSR_AUTH_ORACLE: [0x1566BFA55D95686a823751298533D42651183988](https://uniscan.xyz/address/0x1566BFA55D95686a823751298533D42651183988)

- L2_GOV_RECEIVER: [0x7B8ee8b0fD62662F7FB1ac9e5E6cEAad5195A3bF](https://uniscan.xyz/address/0x7B8ee8b0fD62662F7FB1ac9e5E6cEAad5195A3bF)
- L2_GOV_EXECUTOR: [0xb037C43b433964A2017cd689f535BEb6B0531473](https://uniscan.xyz/address/0xb037C43b433964A2017cd689f535BEb6B0531473)

- L2_PSM: [0x7b42Ed932f26509465F7cE3FAF76FfCe1275312f](https://uniscan.xyz/address/0x7b42Ed932f26509465F7cE3FAF76FfCe1275312f)

- L2_ALM_PROXY: [0x345E368fcCd62266B3f5F37C9a131FD1c39f5869](https://uniscan.xyz/address/0x345E368fcCd62266B3f5F37C9a131FD1c39f5869)
- L2_ALM_CONTROLLER: [0x9B1BEB11CFE05117029a30eb799B6586125321FF](https://uniscan.xyz/address/0x9B1BEB11CFE05117029a30eb799B6586125321FF?tab=contract)
- L2_ALM_RATE_LIMITS: [0x5A1a44D2192Dd1e21efB9caA50E32D0716b35535](https://uniscan.xyz/address/0x5A1a44D2192Dd1e21efB9caA50E32D0716b35535)

> 

Reviewer: ```suggestion
```

##### Initial [Rate Limits](https://docs.spark.fi/dev/spark-liquidity-layer/spark-alm-controller#rate-limits)

###### PSM3

- USDC deposit: **50 million** max, **50 million** slope per day.
- USDC withdrawal: **50 million** max, **50 million** slope per day.
- USDS deposit: **Unlimited**.
- USDS withdrawal: **Unlimited**.
- sUSDS deposit: **Unlimited**.
- sUSDS withdrawal: **Unlimited**.

###### CCTP

- Global: **Unlimited**.
- Unichain to Mainnet: **50 million** max, **25 million** slope per day.
- Mainnet to Unichain: **50 million** max, **25 million** slope per day.

###### Initial Liquidity Bridged to Unichain

- USDS: **100 million USDS**.
- sUSDS: **100 million USDS worth of sUSDS**.

#### Onboard OP Mainnet to the Spark Liquidity Layer

- **Authorization**: [Governance Poll 1254](https://vote.makerdao.com/polling/QmNe8Erm)
- **Proposal**: [Forum Post](https://forum.sky.money/t/may-29-2025-proposed-changes-to-spark-for-upcoming-spell/26372)

Onboard Optimism Mainnet to the Spark Liquidity Layer with the following addresses and parameters.

##### Addresses

> 

Reviewer: ```suggestion
```

- L2_SSR_RECEIVER: [0xE2868095814c2714039b3A9eBEE035B9E2c411E5](https://optimistic.etherscan.io/address/0xE2868095814c2714039b3A9eBEE035B9E2c411E5)
- L2_SSR_AUTH_ORACLE: [0x6E53585449142A5E6D5fC918AE6BEa341dC81C68](https://optimistic.etherscan.io/address/0x6E53585449142A5E6D5fC918AE6BEa341dC81C68)

- L2_GOV_RECEIVER: [0x61Baf0Ce69D23C8318c786e161D1cAc285AA4EA3](https://optimistic.etherscan.io/address/0x61Baf0Ce69D23C8318c786e161D1cAc285AA4EA3)
- L2_GOV_EXECUTOR: [0x205216D89a00FeB2a73273ceecD297BAf89d576d](https://optimistic.etherscan.io/address/0x205216D89a00FeB2a73273ceecD297BAf89d576d)

- L2_PSM: [0xe0F9978b907853F354d79188A3dEfbD41978af62](https://optimistic.etherscan.io/address/0xe0F9978b907853F354d79188A3dEfbD41978af62)

- L2_ALM_PROXY: [0x876664f0c9Ff24D1aa355Ce9f1680AE1A5bf36fB](https://optimistic.etherscan.io/address/0x876664f0c9Ff24D1aa355Ce9f1680AE1A5bf36fB)
- L2_ALM_CONTROLLER: [0x1d54A093b8FDdFcc6fBB411d9Af31D96e034B3D5](https://optimistic.etherscan.io/address/0x1d54A093b8FDdFcc6fBB411d9Af31D96e034B3D5)
- L2_ALM_RATE_LIMITS: [0x6B34A6B84444dC3Fc692821D5d077a1e4927342d](https://optimistic.etherscan.io/address/0x6B34A6B84444dC3Fc692821D5d077a1e4927342d)

> 

Reviewer: ```suggestion
```
